### PR TITLE
✨ Add Share page

### PR DESCRIPTION
**Pull request by AuroraHuang22**
TO-DO:
- [x]  RWD
- [x] Replace user card with [userStats]. (may need to merge my-dashboard to master first)
- [ ] Update copywriting
- [x] Update OG title & description

截圖 2022-09-04 下午11 59 07

### CODE REVIEW

**Comment by nwingt** [comment]
First impression
1. Button [Collected] -> [View Portfolio] 
2. I suggest the share page should have a new route, for separating the logic between details and share, i.e. `/nft/class/:classId/share?referrer=referred`, and redirect `/nft/class/:classId?referrer` to that page

File: src/pages/nft/class/_classId/index.vue [hunk 1]
```diff
@@ -258,8 +237,16 @@ export default {
...
-  async fetch({ route, store }) {
+  async fetch({ route, store, redirect }) {
     const { classId } = route.params;
+    const { referrer } = route.query;
+    if (referrer) {
+      redirect({
+        name: 'nft-class-classId-share',
+        params: { classId },
+        query: { referrer },
+      });
```
Comment (nwingt) [inline src/pages/nft/class/_classId/index.vue:248]: Should add `return;` after redirect to stop further requests

File: src/pages/nft/class/_classId/share/index.vue [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+  async fetch({ route, store }) {
+    const { classId } = route.params;
+    await store.dispatch('fetchNFTMetadata', classId);
+  },
+  async mounted() {
+    await Promise.all([
+      this.updateDisplayNameList(this.iscnOwner),
+      this.updateNFTPurchaseInfo(),
+      this.updateNFTOwners(),
+      this.updateUerInfo(),
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:239]:
```suggestion
      this.updateReferrerInfo(),
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+export default {
+  layout: 'default',
+  filters: {
+    ellipsis,
+  },
+  mixins: [nftMixin, navigationListenerMixin, walletMixin],
+  data() {
+    return {
+      userInfo: null,
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:181]:
```suggestion
      referrerInfo: null,
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+      currentPrice: 0,
+      isCollecting: false,
+    };
+  },
+  computed: {
+    classId() {
+      return this.$route.params.classId;
+    },
+    isUserCivicLiker() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:192]:
```suggestion
    isReferrerCivicLiker() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+      return !!(
+        this.userInfo &&
+        (this.userInfo.isCivicLikerTrial ||
+          this.userInfo.isSubscribedCivicLiker)
+      );
+    },
+    userDisplayName() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:199]:
```suggestion
    referrerDisplayName() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+      return (this.userInfo && this.userInfo.displayName) || this.referrer;
+    },
+    userDescription() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:202]:
```suggestion
    referrerDescription() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+      return this.userInfo && this.userInfo.description;
+    },
+    userAvatar() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:205]:
```suggestion
    referrerAvatar() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+      return (
+        (this.userInfo && this.userInfo.avatar) ||
+        `https://avatars.dicebear.com/api/identicon/${this.referrer}.svg`
+      );
+    },
+    isCreator() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:211]:
```suggestion
    isReferrerTheCreator() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+      return this.referrer === this.iscnOwner;
+    },
+    isCollector() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:214]:
```suggestion
    isReferrerTheCollector() {
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+      </section>
+    </div>
+  </Page>
+</template>
+
+<script>
+import { getLIKEPrice, getAddressLikerIdMinApi } from '~/util/api';
+import { logTrackerEvent } from '~/util/EventLogger';
+import { LIKECOIN_BUTTON_BASE } from '~/constant';
+import { ellipsis } from '~/util/ui';
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:167]:
```suggestion
import { LIKECOIN_BUTTON_BASE } from '~/constant';

import { getLIKEPrice, getAddressLikerIdMinApi } from '~/util/api';
import { logTrackerEvent } from '~/util/EventLogger';
import { ellipsis } from '~/util/ui';
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+    },
+    handleClickNFTDetails() {
+      this.$router.push({
+        name: 'nft-class-classId',
+        params: { classId: this.classId },
+        query: undefined,
+      });
+      logTrackerEvent(this, 'NFT', 'NFTViewDetails(Share)', this.classId, 1);
+    },
+    handleLike() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:299]: Can be refactored and moved to `nftMixin`?

File: src/pages/nft/class/_classId/share/index.vue [hunk 12]
```diff
@@ -0,0 +1,359 @@
...
+      this.referrer !== this.iscnOwner &&
+      !Object.prototype.hasOwnProperty.call(this.ownerInfo, this.referrer)
+    ) {
+      this.invalidToShare();
+      return;
+    }
+    this.isLoading = false;
+  },
+  methods: {
+    async handleCollect() {
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:252]: Can be refactored to `nftMixin`?

File: src/pages/nft/class/_classId/share/index.vue [hunk 13]
```diff
@@ -0,0 +1,359 @@
...
+            'desktop:col-span-2',
+          ]"
+        >
+          <Label preset="h3" class="text-like-green font-600" :text="shareTitle" />
+          <div class="w-[480px]">
+            <div
+              class="rounded-[18px] p-[3px] bg-cover"
+              style="
+                background-image: url('/images/gradient/like-gradient-lighter-blur.svg');
+              "
```
Comment (nwingt) [inline src/pages/nft/class/_classId/share/index.vue:94]:
```suggestion
              class="rounded-[18px] p-[3px] bg-cover bg-[url('/images/gradient/like-gradient-lighter-blur.svg')]"
```

File: src/pages/nft/class/_classId/share/index.vue [hunk 14]
```diff
@@ -0,0 +1,359 @@
...
+    await Promise.all([
+      this.updateDisplayNameList(this.iscnOwner),
+      this.updateNFTPurchaseInfo(),
+      this.updateNFTOwners(),
+      this.updateUerInfo(),
+    ]);
+    if (
+      this.referrer &&
+      this.referrer !== this.iscnOwner &&
+      !Object.prototype.hasOwnProperty.call(this.ownerInfo, this.referrer)
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply src/pages/nft/class/_classId/share/index.vue:244]: I wonder too 😂

**Review by nwingt — APPROVED** [review]

**Pull request merged**